src/darwin/Framework/CHIP/CHIPPersistentStorageDelegateBridge.mm

```diff
@@ -112,43 +114,50 @@
     return error;
 }
 
-void CHIPPersistentStorageDelegateBridge::AsyncSetKeyValue(const char * key, const char * value)
+CHIP_ERROR CHIPPersistentStorageDelegateBridge::SyncSetKeyValue(const char * key, const void * value, uint16_t size)
 {
+    std::string base64Value = StringToBase64(std::string(static_cast<const char *>(value), size));
+
     NSString * keyString = [NSString stringWithUTF8String:key];
-    NSString * valueString = [NSString stringWithUTF8String:value];
-    dispatch_async(mWorkQueue, ^{
-        NSLog(@"PersistentStorageDelegate Set Key %@, Value %@", keyString, valueString);
+    NSString * valueString = [NSString stringWithUTF8String:base64Value.c_str()];
 
-        id<CHIPPersistentStorageDelegate> strongDelegate = mDelegate;
-        if (strongDelegate && mQueue) {
-            dispatch_async(mQueue, ^{
-                [strongDelegate CHIPSetKeyValue:keyString value:valueString handler:mSetStatusHandler];
-            });
-        } else {
-            [mDefaultPersistentStorage setObject:valueString forKey:keyString];
-            if (mSetStatusHandler) {
-                mSetStatusHandler(keyString, [CHIPError errorForCHIPErrorCode:0]);
-            }
-        }
-    });
+    NSLog(@"PersistentStorageDelegate Set Key %@", keyString);
+
+    id<CHIPPersistentStorageDelegate> strongDelegate = mDelegate;
+    if (strongDelegate && mQueue) {
+        dispatch_sync(mQueue, ^{
+            [strongDelegate CHIPSetKeyValue:keyString value:valueString];
+        });
+    } else {
+        [mDefaultPersistentStorage setObject:valueString forKey:keyString];
+    }
+
+    // TODO: ideally the error from the dispatch should be returned
```

Comment:
You you use a block variable here and update it when the delegate returns a result. We can require that the objc storage APIs return errors.

Reply:
The reason I was a bit loose here is that we have a mac KVS implementation that does sync calls and returns error codes. I plan to switch out the implementation to have a KVS implementation everywhere so we have a single storage abstraction.

Reply:
I think we want to leave the storage APIs open for consumers of the SDK to override.

Reply:
I think we want to leave the storage APIs open for consumers of the SDK to override.

Reply:
I believe key-value-store is overridable or alternatively the PersistentStorageDelegate itself is.

I wonder if we should have a side discussion on how to provide a single consistent override mechanism and what that is (I was always assuming it would be part of 'platform startup code' but having KVS as a singleton kind of defeats this)